Library/Homebrew/cask/audit.rb

```diff
@@ -76,66 +76,37 @@ def errors
       @errors ||= []
     end
 
-    def warnings
-      @warnings ||= []
-    end
-
     sig { returns(T::Boolean) }
     def errors?
       errors.any?
     end
 
-    sig { returns(T::Boolean) }
-    def warnings?
-      warnings.any?
-    end
-
     sig { returns(T::Boolean) }
     def success?
-      !(errors? || warnings?)
+      !errors?
     end
 
-    sig { params(message: T.nilable(String), location: T.nilable(String)).void }
-    def add_error(message, location: nil)
-      errors << ({ message: message, location: location })
-    end
+    sig { params(message: T.nilable(String), location: T.nilable(String), warning: T::Boolean).void }
+    def add_error(message, location: nil, warning: false)
```

Comment:
IMO
```suggestion
    def add_error(message, location: nil, strict_only: false)
```
or
```suggestion
    def add_error(message, location: nil, stricty: false)
```
would make more sense?